### PR TITLE
Installation Flame 0.17.1 parallel build

### PR DESCRIPTION
**Pull request by desmfryan**
Installation of Flame framework - agent based model. Parallel build. Test section included.

### CODE REVIEW

**Comment by willfurnass** [comment]
Assign a reviewer if this is ready for review.

**Comment by dmrogers75** [comment]
Hello Will. Assigned on behalf of Des! Cheers, David

File: iceberg/software/apps/flame.rst [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+.. code-block:: bash
+
+   #!/bin/bash
+   #$ -pe openmpi-ib no_cores
+   #$ -cwd
+   #$ -j y
+   #$ -m bea
+   #$ -M email_address
+   module load apps/flame
```
Comment (willfurnass) [inline iceberg/software/apps/flame.rst]: `module load apps/gcc/4.8.2/flame/0.17.1`?

File: iceberg/software/apps/flame.rst [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+Usage
+-----
+
+Flame can be activated using the module file::
+
+    module load apps/gcc/4.8.2/flame/0.17.1
+
+
+Note: The module file also loads libtool, libmboard, and the compilers gcc/4.8.3 & openmpi/1.8.8. Libmboard is the communication library used by Flame simulation programs.
```
Comment (willfurnass) [inline iceberg/software/apps/flame.rst]: Link to pages for other applications?

File: iceberg/software/install_scripts/apps/gcc/4.8.2/flame/0.17.1/install_flame.sh [hunk 3]
```diff
@@ -0,0 +1,218 @@
...
+
+cd $version
+
+#note in Makefile.tmpl
+#set LIBMBOARD_DIR=/usr/local/packages6/apps/gcc/4.8.2/flame/0.17.1-openmpi-1.8.8/libmboard/0.3.1
+make
+#make install prefix=$prefix
+cp -r * $prefix
+
+bash
```
Comment (willfurnass) [inline iceberg/software/install_scripts/apps/gcc/4.8.2/flame/0.17.1/install_flame.sh]: ?

File: iceberg/software/modulefiles/apps/gcc/4.8.2/flame/0.17.1 [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+##
+
+proc ModulesHelp { } {
+        global version
+
+        puts stderr "   Makes Flame (xparser) Version 0.17.1 available for use"
+}
+
+module-whatis   "Makes Flame (xparser) Version 0.17.1 available"
+
```
Comment (willfurnass) [inline iceberg/software/modulefiles/apps/gcc/4.8.2/flame/0.17.1]: Comment that also makes particular GCC and OpenMPI available?

File: iceberg/software/apps/flame.rst [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+
+The FLAME framework is an enabling tool to create agent-based models that can be run on high performance computers (HPCs). Models are created based upon a model of computation called (extended finite) state machines. By defining agent-based models in this way the FLAME framework can automatically generate simulation programs that can run models efficiently on HPCs. The behaviour model is based upon state machines which are composed of a number of states with transition functions between those states. There is a single start state and by traversing states using the transition functions the machine executes the functions until it reaches an end state. This happens to each agent/machine as one time step or iteration is completed
+
+
+Usage
+-----
+
+Flame can be activated using the module file::
+
+    module load apps/gcc/4.8.2/flame/0.17.1
```
Comment (willfurnass) [inline iceberg/software/apps/flame.rst]: May want to install same flame version built with different OpenMPI version in future - better to name module file `apps/gcc/4.8.2/flame/0.17.1-openmpi-1.8.8` (in keeping with install prefix)?

File: iceberg/software/apps/flame.rst [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+   #$ -cwd
+   #$ -j y
+   #$ -m bea
+   #$ -M email_address
+   module load apps/flame
+   make clean
+   cp /usr/local/packages6/apps/gcc/4.8.2/flame/0.17.1-openmpi-1.8.8/xparser/0.17.1/*.tmpl .
+   xparser model.xml -p
+   make
+   mpirun -np no_cores ./main no_its its/0.xml
```
Comment (willfurnass) [inline iceberg/software/apps/flame.rst]: Cannot find `its` subdir when try to run `model_01` example

File: iceberg/software/apps/flame.rst (hunk 5 above)
Comment (desmfryan) [inline iceberg/software/apps/flame.rst]: Done.

File: iceberg/software/apps/flame.rst (hunk 2 above)
Comment (desmfryan) [inline iceberg/software/apps/flame.rst]: I believe that the main link to Flame website addressed this.

File: iceberg/software/apps/flame.rst (hunk 1 above)
Comment (desmfryan) [inline iceberg/software/apps/flame.rst]: Done.

File: iceberg/software/apps/flame.rst (hunk 6 above)
Comment (desmfryan) [inline iceberg/software/apps/flame.rst]: Added a comment that user is required to create an 'its' directory, and copy 0.xml into it (for each model_0*.

File: iceberg/software/install_scripts/apps/gcc/4.8.2/flame/0.17.1/install_flame.sh (hunk 3 above)
Comment (desmfryan) [inline iceberg/software/install_scripts/apps/gcc/4.8.2/flame/0.17.1/install_flame.sh]: Sorted.

File: iceberg/software/modulefiles/apps/gcc/4.8.2/flame/0.17.1 (hunk 4 above)
Comment (desmfryan) [inline iceberg/software/modulefiles/apps/gcc/4.8.2/flame/0.17.1]: Done.

**Comment by desmfryan** [comment]
Hi Will

Despite all the effort I have spotted a 'typo'. In the code section of the flame.rst file the module load has not been updated.

It is currently:
module load apps/flame

whereas as it should be:
module load apps/gcc/4.8.2/flame/0.17.1

Should I make the changes and make another pull request?

Cheers,

Des

**Comment by willfurnass** [comment]
@desmfryan Yes

**Pull request merged**